src/main/java/org/springframework/data/elasticsearch/core/convert/MappingElasticsearchConverter.java

```diff
@@ -220,6 +218,17 @@ public void afterPropertiesSet() {
 
 	}
 
+	private SeqNoPrimaryTerm fillSeqNoPrimaryTermFromDocument(Document document) {
+		SeqNoPrimaryTerm seqNoPrimaryTerm = new SeqNoPrimaryTerm();
+		if (document.hasSeqNo()) {
+			seqNoPrimaryTerm.setSequenceNumber(document.getSeqNo());
+		}
+		if (document.hasPrimaryTerm()) {
+			seqNoPrimaryTerm.setPrimaryTerm(document.getPrimaryTerm());
+		}
+		return seqNoPrimaryTerm;
+	}
```

Comment:
would not be needed, when the factory method `SeqNoPrimaryTerm.of(...)` is used like above

Reply:
This logic has been moved to `SeqNoPrimaryTerm.ofAssigned()` which returns an `Optional`.